api/users.rb

```diff
@@ -156,3 +156,12 @@
   user.replace_username_in_all_content(params["new_username"])
   user.save
 end
+
+post "#{APIPREFIX}/users/:course_id/update_stats" do |course_id|
+  active_contents = Content.where(anonymous: false, anonymous_to_peers: false, course_id: course_id)
+  for content in active_contents do
+    user = User.find_by(username: content.author_username)
+    build_course_stats_for_user(user, course_id)
+  end
+  course_id
+end
```

Comment:
I think this code can be optimised quite a bit with a few changes:
1. All we need from the content are distinct user ids. Currently, each user might get multiple stats updates since they will have multiple content entries. You can use something like: `Content.where(anonymous: false, anonymous_to_peers: false, course_id: course_id).distinct(:author_id)`. This should give you all the distinct author ids involved
2. While `build_course_stats_for_user` currently needs a user object, there is no reason for this to be the case. All it needs is the `external_id`, which should also match the `author_id` you get from above. So, you can modify the `build_course_stats_for_user` function to take just the user id and avoid the lookup here. 
3. A course might have thousands or tens of thousands of users who've participated in discussion. This should probably be a background job. I think the installed `delayed_job` gem should help here but I'm not very familiar.

Reply:
@xitij2000 thank, it make sense. 
Regarding the point 2 , the `build_course_stats_for_user` function needs a `user` object https://github.com/openedx/cs_comments_service/blob/master/models/comment.rb#L258, it seems we need to lookup for the user object anyway , what do you think ?

Reply:
It seems so, I guess I missed that! I think you can focus on 1. for now and I'll see if 3 is feasible.

Reply:
Point 1 was already done, I am working on 3. It should be feasible.

Reply:
@xitij2000 I changed `build_course_stats_for_user` to be a background job.